src/lib.rs

```diff
@@ -35,41 +35,52 @@ impl ToPtr for CString {
 }
 
 impl Flux {
-    fn open(uri: &str, flags: u32) -> Result<Flux> {
+    pub fn open(uri: &str, flags: u32) -> Result<Flux> {
+        // N.B.: need to create the CString outside of the if statement since
+        // the pointer returned by as_ptr has no lifetime information.
+        let new_uri = CString::new(uri)?;
         let p = if uri.len() == 0 {
             std::ptr::null()
         } else {
-            CString::new(uri)?.as_ptr()
+            new_uri.as_ptr()
         };
         unsafe { flux_sys::flux_open(p, flags as i32) }
             .flux_check()
             .map(|f| Flux { handle: f })
     }
-    fn attr_get(self: &mut Self, name: &str) -> Result<&str> {
+    pub fn attr_get(self: &mut Self, name: &str) -> Result<&str> {
         let s = CString::new(name)?;
         unsafe {
             let cstr_ptr = flux_sys::flux_attr_get(self.handle, s.as_ptr()).flux_check()?;
             Ok(CStr::from_ptr(cstr_ptr).to_str()?)
         }
     }
-    fn service_register(self: &mut Self, name: &str) -> Result<FluxFuture> {
+    pub fn service_register(self: &mut Self, name: &str) -> Result<FluxFuture> {
         let s = CString::new(name)?;
         Ok(unsafe {
             FluxFuture::from_ptr(
                 flux_sys::flux_service_register(self.handle, s.as_ptr()).flux_check()?,
             )
         })
     }
-    fn kvs_lookup(
+    pub fn kvs_lookup(
         self: &mut Self,
         flags: flux_sys::kvs_op::Type,
         key: &str,
     ) -> Result<FluxKvsFuture> {
         let k = CString::new(key)?;
-        let fut_ptr = unsafe { flux_sys::flux_kvs_lookup(self.handle, std::ptr::null(), flags as i32, k.as_ptr()) }
-            .flux_check()?;
+        let fut_ptr = unsafe {
+            flux_sys::flux_kvs_lookup(self.handle, std::ptr::null(), flags as i32, k.as_ptr())
+        }
+        .flux_check()?;
```

Comment:
Yeah, it looks like it is.  I tried merging it in with the previous line then re-running the formatter and got the same thing. I agree it looks very weird as a singleton.  I like their new-line-per-method format when there are more chained methods, but for one it seems odd.

Reply:
Oh! That reminds me that I should add cargo format to the CI.

Reply:
That's a good idea.  I think it's bugging me because the thing on the previous line is a curly-brace, which is fine in rust but *very very odd* in C++.  I suppose it can come up, but man does it make my WTF sense tingle.  Either way going to resolve this, whether I like it in all places or not rustfmt is the way to go. 👍